devices/IAnalogSensorToIWear/src/IAnalogSensorToIWear.cpp

```diff
@@ -405,7 +419,8 @@ bool IAnalogSensorToIWear::open(yarp::os::Searchable& config)
     // INITIALIZE THE DEVICE
     // =====================
 
-    handler.buffer.resize(pImpl->options.numberOfChannels);
+    pImpl->handler = new Impl::IAnalogSensorHandler();
```

Comment:
I see many possible problems in using this object as raw pointer. I would switch to a `std::shared_ptr` so that the ownership of the handler can be shared safely with all the sensor classes.

Reply:
It is created in the device class and shared with all the sensor objects

Reply:
You are right, but as the device class as a shared_ptr toward the sensor object, how do we avoid that this creates a cyclic structure? See http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rr-weak_ptr .

Reply:
> If the pointer is never shared, why not use a `std::unique_ptr` ?

The pointer is shared with the sensors implementation (https://github.com/lrapetti/wearables/blob/fix-IAnalogSensorToIWear/devices/IAnalogSensorToIWear/src/IAnalogSensorToIWear.cpp#L241)

Reply:
See also https://www.learncpp.com/cpp-tutorial/15-7-circular-dependency-issues-with-stdshared_ptr-and-stdweak_ptr/ .

Reply:
I had a broader look of the class. We can clean up the `handler` usage as follows:

1. Remove it from the Impl class
1. Create a temporary class in the attach()
1. Add a third parameter to `allocateSensor()` with the handler (passed by copy)

In this way we can avoid dynamic allocation (which in any case is very small) and we have for free the deletion of the handler object in the `detach()` method since it will be destroyed together with the sensor.

Reply:
I have implemented and tested the solution proposed by @diegoferigo with https://github.com/robotology/wearables/pull/52/commits/6983116fcf7969c18bbb319b65cda0334e782c17